### ISSUE TITLE
Proposal: Add SelectedIndex property to NavigationView

### ISSUE THREAD

**Issue by Jaiganeshkumaran** [microsoft/microsoft-ui-xaml#5009]
# Proposal: Add SelectedIndex property to NavigationView
Today NavigationView only offers a SelectedItem property unlike other XAML controls. I think it would be better if NavigationView has a SelectedIndex property. Since some items can be in the footer menu items group and some in the main menu items group, the property should calculate the logical index with MenuItems first, then FooterMenuItems, then the settings item.

## Summary

Add a int32_t property named SelectedIndex. It should be -1 when nothing's selected and the actual index of the selected item if it's in the menu items group and if it's in the footer menu items the index should MenuItems.Size + the index inside footer menu items group.

## Rationale

* Makes it consistent with other XAML controls

## Scope

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to change the selected item based on the logical index | Must |

**Comment by chingucoding** [comment]
I don't think that this makes much sense to add, selection in NavigationView is a lot more like TreeView, not like ListView or GridView. The Items can be nested and at this point a selected index falls apart.

**Comment by mdtauk** [comment]
How would a selected index work with the child items which are rendered in a flyout?

How does this index handle re-arranging items?

What about Bottom Items, or custom elements in the bottom/right sections?

**Comment by Jaiganeshkumaran** [comment]
> How would a selected index work with the child items which are rendered in a flyout?
> 
Those items could be treated as separate elements.

> How does this index handle re-arranging items?
> 
If the items are rearranged, the index would be changed. It won't be really saved somewhere but only calculated when you ask for it.

> What about Bottom Items, or custom elements in the bottom/right sections?
If it's part of the footer menu items group, SelectedIndex should return the index + the size of the menu items group which are displayed first on top or right. Other unselectable elements should also be included in index calculation however setting it wouldn't really do anything

**Comment by Jaiganeshkumaran** [comment]
> I don't think that this makes much sense to add, selection in NavigationView is a lot more like TreeView, not like ListView or GridView. The Items can be nested and at this point a selected index falls apart.

Most apps use NavigationView for static navigation so there should be an easy to way to select a specific element using an unified index instead of separately searching them in MenuItems, then FooterMenuItems

**Comment by mdtauk** [comment]
> > I don't think that this makes much sense to add, selection in NavigationView is a lot more like TreeView, not like ListView or GridView. The Items can be nested and at this point a selected index falls apart.
> 
> Most apps use NavigationView for static navigation so there should be an easy to way to select a specific element using an unified index instead of separately searching them in MenuItems, then FooterMenuItems

You can't really include a feature that only works in certain circumstances, and breaks anytime someone deviates slightly from the use case you expect.

If you want to include SelectedIndex, it needs to work with all the variations of the control - so you need to answer the questions brought up, plan it thoroughly, and then figure out how to implement it.

**Comment by chingucoding** [comment]
> Most apps use NavigationView for static navigation so there should be an easy to way to select a specific element using an unified index instead of separately searching them in MenuItems, then FooterMenuItems

And how do you know what the correct selected index is? If you want to select the "Page A item", how do you know what it's index is? Do you just hardcode the index? Than you could also just use x:Name and reference that way. And if you want to access it from the outside, provide a selected index mapping to the elements.

> You can't really include a feature that only works in certain circumstances, and breaks anytime someone deviates slightly from the use case you expect.
> 
> If you want to include SelectedIndex, it needs to work with all the variations of the control - so you need to answer the questions brought up, plan it thoroughly, and then figure out how to implement it.

💯

**Comment by Jaiganeshkumaran** [comment]
> > Most apps use NavigationView for static navigation so there should be an easy to way to select a specific element using an unified index instead of separately searching them in MenuItems, then FooterMenuItems
> 
> And how do you know what the correct selected index is? If you want to select the "Page A item", how do you know what it's index is? Do you just hardcode the index? Than you could also just use x:Name and reference that way. And if you want to access it from the outside, provide a selected index mapping to the elements.
> 
Yes for the default I would hardcode 0 or some other index instead of manually setting it through code behind in the loaded and keeping track whether it's actually the first time or not. Also x:Bind to controls with x:Name doesn't seem to work in C++ projects  so I can't use that.

**Comment by Jaiganeshkumaran** [comment]
The items are null in the beginning so setting it wouldn't actually work. You have to set it after the first Loaded event

**Comment by chingucoding** [comment]
> > > Most apps use NavigationView for static navigation so there should be an easy to way to select a specific element using an unified index instead of separately searching them in MenuItems, then FooterMenuItems
> > 
> > 
> > And how do you know what the correct selected index is? If you want to select the "Page A item", how do you know what it's index is? Do you just hardcode the index? Than you could also just use x:Name and reference that way. And if you want to access it from the outside, provide a selected index mapping to the elements.
> 
> Yes for the default I would hardcode 0 or some other index instead of manually setting it through code behind in the loaded and keeping track whether it's actually the first time or not. Also x:Bind to controls with x:Name doesn't seem to work in C++ projects so I can't use that.

You haven't actually addressed how you would know if I clicked on the "icons section" or the "colors section" of your app (assuming that are areas of your navigation). Do you just hardcode the value what index is which section?

I'm having a hard time understanding what problem exactly you are trying to solve with that property, if you can elaborate what problem this property would solve it might be easier to see if it's a good idea or not.
